corporate/views/support.py

```diff
@@ -0,0 +1,68 @@
+from urllib.parse import urlencode, urljoin, urlunsplit
+
+from django import forms
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.urls import reverse
+
+from zerver.decorator import zulip_login_required
+from zerver.lib.request import has_request_variables
+from zerver.lib.send_email import FromAddress, send_email
+from zerver.models import get_realm
+
+
+class SupportRequestForm(forms.Form):
+    MAX_SUBJECT_LENGTH = 100
+    request_subject = forms.CharField(max_length=MAX_SUBJECT_LENGTH)
+    request_message = forms.CharField(widget=forms.Textarea)
+
+
+@zulip_login_required
+@has_request_variables
+def support_request(request: HttpRequest) -> HttpResponse:
+    user = request.user
+    assert user.is_authenticated
+
+    context = {
+        "email": user.delivery_email,
+        "realm_name": user.realm.name,
+        "MAX_SUBJECT_LENGTH": SupportRequestForm.MAX_SUBJECT_LENGTH,
+    }
+
+    if request.POST:
+        post_data = request.POST.copy()
+        form = SupportRequestForm(post_data)
+
+        if form.is_valid():
+            support_realm_uri = get_realm(settings.STAFF_SUBDOMAIN).uri
+            support_url = urljoin(
+                support_realm_uri,
+                urlunsplit(
+                    ("", "", reverse("support"), urlencode({"q": user.realm.string_id}), "")
+                ),
+            )
```

Comment:
This blob should probably be a function, you can put it in `corporate/lib/support.py` if you don't have a more obvious place, to share with the `upgrade.py` code path.